compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -364,6 +367,21 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
         self.insert(Value::Aggregate(AggregateTy::Tuple, VariantIdx::ZERO, values))
     }
 
+    fn insert_deref(&mut self, value: VnIndex) -> VnIndex {
+        let value = self.insert(Value::Projection(value, ProjectionElem::Deref));
+        self.derefs.push(value);
+        value
+    }
+
+    fn invalidate_derefs(&mut self) {
+        let mut derefs = Vec::new();
+        std::mem::swap(&mut derefs, &mut self.derefs);
+        for deref in derefs {
```

Comment:
Use

```rust
for deref in self.derefs.drain(..) {
```

There are no borrowing issues (if there were, `for deref in std::mem::take(&mut self.derefs)` would work), and it signals the intent here better.

Reply:
Ah, thanks!

Reply:
I used `std::mem::take(&mut self.derefs)`. `next_opaque`'s argument is `&mut self`.